sdk/monitor/monitor-ingestion/src/utils/concurrentPoolHelper.ts

```diff
@@ -23,6 +26,10 @@ export async function concurrentRun<T>(
     if (promises.length === maxConcurrency) {
       await Promise.race(promises);
     }
+    if (abortSignal?.aborted) {
```

Comment:
So if I understand this correctly, since we pass the same abortSignal to the underlying upload operation, when abort is called it will cancel any pending requests that are still in-flight and then this bit of code will prevent us from scheduling more and let us forward the rejections back to the caller?

Clever!